contracts/src/system/OnBlockEndCallback.sol

```diff
@@ -0,0 +1,8 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+import "./Transaction.sol";
+
+
+interface OnBlockEndCallback {
+    function onBlockEnd(Structs.Transaction[] calldata transactions) external;
```

Comment:
It would run up the gas cost significantly as it would be making a new call and state modification for each tx. Obiously this is on the L2 and its free for us, but I think performance won't be great under big load. A lot of stuff gets initialized when making a call behind the scenes. Also it's better to have the whole batch context, in case in the future we decide to have more complex rewarding.